karapace/protobuf/schema.py

```diff
@@ -106,6 +109,49 @@ def option_element_string(option: OptionElement) -> str:
     return f"option {result};\n"
 
 
+@dataclasses.dataclass(frozen=True)
+class UsedType:
+    parent_object_qualified_name: str
+    used_attribute_type: str
+
+
+@dataclasses.dataclass(frozen=True)
+class TypeTree:
+    token: str
+    children: List["TypeTree"]
```

Comment:
suggestion (non-blocking): It would be comforting to make this immutable, but potentially requires a bit of refactoring of the constructing algorithm.

Reply:
yeah, you are right 😄 was already on plan before merging, I've said that to @jjaakola-aiven before the review 😄 happy that also you noticed it, doing it now

Reply:
ok, experiment done and wasn't easy to rewrite the parser. I've tried to explain a little bit a possible follow-up, wdyt?

Reply:
Yeah, this is not a blocker, thanks for exploring the options 🙏 I'll mark as resolved.